cmd/dashboard.go

```diff
@@ -76,19 +83,81 @@ var dashboardCmd = &cobra.Command{
 }
 
 func runPortforward(podName string, localPort int) error {
-	cmd, err := exec.LookPath("kubectl")
+	stopChannel := make(chan struct{}, 1)
+	readyChannel := make(chan struct{})
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+	defer signal.Stop(signals)
+
+	go func() {
+		<-signals
+		if stopChannel != nil {
+			close(stopChannel)
+		}
+	}()
+
+	// Open the Dashboard in a browser when the port-forward is established
+	go func() {
+		<-readyChannel
+		openInBrowser(fmt.Sprintf("http://localhost:%d", localPort))
+	}()
+
+	fw, err := newPortforwarder(podName, localPort, stopChannel, readyChannel)
 	if err != nil {
 		return err
 	}
-	args := []string{"kubectl", "--namespace", ingressNamespace, "port-forward", podName, fmt.Sprintf("%d:80", localPort)}
+	return fw.ForwardPorts()
+}
 
-	env := os.Environ()
+func newPortforwarder(podName string, localPort int, stopChannel, readyChannel chan struct{}) (*portforward.PortForwarder, error) {
+	config, err := restClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := portforwardReqURL(config, podName)
+	if err != nil {
+		return nil, err
+	}
+
+	dialer, err := remotecommand.NewExecutor(config, "POST", url)
+	if err != nil {
+		return nil, err
+	}
+
+	ports := []string{fmt.Sprintf("%d:80", localPort)}
+
+	return portforward.New(dialer, ports, stopChannel, readyChannel, os.Stdout, os.Stderr)
+}
+
+func restClientConfig() (*rest.Config, error) {
+	config, err := buildOutOfClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	config.APIPath = "/api"
+	config.GroupVersion = &v1.SchemeGroupVersion
+	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
+	return config, nil
+}
+
+func portforwardReqURL(config *rest.Config, podName string) (*url.URL, error) {
+	restClient, err := rest.RESTClientFor(config)
+	if err != nil {
+		return nil, err
+	}
 
-	openInBrowser(fmt.Sprintf("http://localhost:%d", localPort))
-	return syscall.Exec(cmd, args, env)
+	req := restClient.Post().
+		Resource("pods").
+		Namespace(ingressNamespace).
```

Comment:
the "ingressNamespace" seems confusing, I know that it points to `kubeapps` but it would be probably better if you make that var a parameter and call it `podNamespace`

Reply:
How about `ingressPodNamespace`?

Reply:
you would be unnecessarily limiting the scope of the function to be used with `ingress` pods, isn't it? As far as I can see you can use this to obtain the URL for any kind of pod.

Reply:
I feel like we're trying to generalise this for no reason. Seems like a pre-optimisation, what other port-forward calls do we expect to make?

Reply:
probably not, it is a generic suggestion

Reply:
OK, unless you feel strongly about it, I think we can leave as-is. If we want to use this in more places in the future we can move it out of this file and add a parameter. Sound good?